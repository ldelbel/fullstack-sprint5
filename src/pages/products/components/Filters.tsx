import { Filter } from "../../../types";

interface IFilterItem {
  label: string;
}

interface IFilters {
  filters: Filter[];
}

function FilterItem({ label }: IFilterItem) {
  return (
    <li className="filters__item">
      <span className="filters__label">{label}</span>
      <img className="filters__img" src="assets/filter.svg" alt="filtro" />
    </li>
  );
}

function Filters({ filters }: IFilters) {
  return (
    <section className="main__filters filters">
      <ul className="filters__list">
        {filters.map((f) => (
          <FilterItem key={f.id} label={f.label} />
        ))}
      </ul>
    </section>
  );
}

export default Filters;
