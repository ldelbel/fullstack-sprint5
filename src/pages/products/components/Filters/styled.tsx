import styled from "styled-components";

export const StyledFilters = styled.section`
    border-top: 1px solid #e3e3e3;
    border-bottom: 1px solid #e3e3e3;

  @media (max-width: 1200px) {
      display: none;
  }

  .filters__list {
    display: flex;
  }


`;
