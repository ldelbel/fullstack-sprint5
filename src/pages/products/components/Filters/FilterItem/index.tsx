import { StyledFilterItem } from "./styled";

interface IFilterItem {
  label: string;
}

export function FilterItem({ label }: IFilterItem) {
  return (
    <StyledFilterItem>
      <span className="filters__label">{label}</span>
      <img className="filters__img" src="assets/filter.svg" alt="filtro" />
    </StyledFilterItem>
  );
}
