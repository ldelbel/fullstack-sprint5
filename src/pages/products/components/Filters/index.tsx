import { Filter } from "../../../../types";
import { FilterItem } from "./FilterItem";
import { StyledFilters } from "./styled";

interface IFilters {
  filters: Filter[];
}

function Filters({ filters }: IFilters) {
  return (
    <StyledFilters>
      <ul className="filters__list">
        {filters.map((f) => (
          <FilterItem key={f.id} label={f.label} />
        ))}
      </ul>
    </StyledFilters>
  );
}

export default Filters;
