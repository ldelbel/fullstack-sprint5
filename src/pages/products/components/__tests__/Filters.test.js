import React from "react";
import { render } from "@testing-library/react";
import Filters from "../Filters";

const filters = [
  {
    id: "size",
    label: "Tamanho",
  },
  {
    id: "color",
    label: "Cor",
  },
];

describe("Filters component", () => {

  it("should render filters", () => {
    const { getByRole } = render(<Filters filters={filters} />);
    expect(getByRole("list")).toHaveClass("filters__list");
  });

  it("should match snapshot", () => {
    const { container } = render(<Filters filters={filters} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  
});
