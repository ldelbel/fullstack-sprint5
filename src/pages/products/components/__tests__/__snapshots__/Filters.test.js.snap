// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component should match snapshot 1`] = `
<section
  class="sc-gtsrHT dZnsuN"
>
  <ul
    class="filters__list"
  >
    <div
      class="sc-bdnxRM gbWXzX"
    >
      <span
        class="filters__label"
      >
        Tamanho
      </span>
      <img
        alt="filtro"
        class="filters__img"
        src="assets/filter.svg"
      />
    </div>
    <div
      class="sc-bdnxRM gbWXzX"
    >
      <span
        class="filters__label"
      >
        Cor
      </span>
      <img
        alt="filtro"
        class="filters__img"
        src="assets/filter.svg"
      />
    </div>
  </ul>
</section>
`;
